lms/djangoapps/mobile_api/tests/test_middleware.py

```diff
@@ -0,0 +1,147 @@
+"""
+tests for version based app upgrade middleware
+"""
+from datetime import datetime
+import ddt
+from django.core.cache import cache, caches
+from django.http import HttpRequest, HttpResponse
+from django.test import TestCase
+import mock
+from pytz import UTC
+from mobile_api.middleware import AppVersionUpgrade
+from mobile_api.models import AppVersionConfig
+from request_cache.middleware import RequestCache
+
+
+@ddt.ddt
+class TestAppVersionUpgradeMiddleware(TestCase):
+    """ Tests for version based app upgrade middleware """
```

Comment:
Nit: The recent convention we use here is to put the docstring on multiple lines, and _not_ use a single line docstring.